ttnn/cpp/ttnn/operations/transformer/sdpa/sdpa.cpp

```diff
@@ -68,4 +69,59 @@ ttnn::Tensor ExecuteScaledDotProductAttention::invoke(
         compute_kernel_config);
 }
 
+ttnn::Tensor ExecuteChunkedScaledDotProductAttention::invoke(
+    uint8_t queue_id,
+    const ttnn::Tensor& input_tensor_q,
+    const ttnn::Tensor& input_tensor_k,
+    const ttnn::Tensor& input_tensor_v,
+    const ttnn::Tensor& page_table_tensor,
+    int64_t chunk_start_idx,
+    std::optional<float> scale,
+    const std::optional<MemoryConfig>& memory_config,
+    std::optional<SDPAProgramConfig> program_config,
+    std::optional<DeviceComputeKernelConfig> compute_kernel_config) {
+    auto arch = input_tensor_q.storage_type() == StorageType::DEVICE
+                    ? input_tensor_q.device()->arch()
+                    : ttnn::operations::experimental::auto_format::AutoFormat::GetDefaultDevice()->arch();
+    auto kernel_config_val = init_device_compute_kernel_config(
+        input_tensor_q.device()->arch(), compute_kernel_config, MathFidelity::HiFi2, true, false, false);
+
+    return operation::run(
+               ScaledDotProductAttention{
+                   .scale = scale,
+                   .output_mem_config = memory_config.value_or(operation::DEFAULT_OUTPUT_MEMORY_CONFIG),
+                   .program_config = program_config,
+                   .is_causal = true,  // Always causal for chunked version
+                   .chunk_start_idx = chunk_start_idx,
+                   .compute_kernel_config = kernel_config_val},
+               {input_tensor_q, input_tensor_k, input_tensor_v},
+               {std::nullopt, page_table_tensor},  // No attention mask - handled internally based on chunk_start_idx
```

Comment:
page table tensor should be second optional tensor. First optional is attention mask